plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveDecimalToVarcharCoercion.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static io.trino.tpch.TpchTable.NATION;
+
+public class TestHiveDecimalToVarcharCoercion
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setInitialTables(ImmutableList.of(NATION))
+                .build();
+    }
+
+    @Test
+    public void testShortDecimalToVarcharCoercion()
+    {
+        String tableName = "test_short_decimal_to_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(10,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
```

Comment:
I have tried your test with a different file format

```
        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(23,3), col2 int, col3 int) with ( FORMAT = 'JSON', partitioned_by = ARRAY['col3'])");
        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.13', 1, 1)");
```

and got a different result from `PARQUET` testing:
```
        String tableName = "test_long_decimal_to_varchar_coercion_" + randomTableSuffix();
        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(23,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.130', 1, 1)");
```

Notice `100.13` for JSON and `100.130` for Parquet.

Reply:
When a partition has decimal and table has varchar,
does Hive's eventual result depend on the partition's file format? 🤯

Reply:
I did some extra checking and realised that for `JSON` data, the `HiveRecordCursor` is employed and not the `HivePageSource`.

There is no coercing done as far as I see in the `HiveRecordCursor`.